machine-runner/src/design/protocol.ts

```diff
@@ -61,16 +54,17 @@ export namespace SwarmProtocol {
    */
   export const make = <
     SwarmProtocolName extends string,
-    RegisteredEventsFactoriesTuple extends MachineEvent.Factory.ReadonlyNonZeroTuple,
+    InitialEventFactoriesTuple extends MachineEvent.Factory.ReadonlyNonZeroTuple,
   >(
     swarmName: SwarmProtocolName,
-    registeredEventFactories: RegisteredEventsFactoriesTuple,
-  ): SwarmProtocol<SwarmProtocolName, RegisteredEventsFactoriesTuple> => {
-    const tag = Tag<MachineEvent.Factory.ReduceToEvent<RegisteredEventsFactoriesTuple>>(swarmName)
+    registeredEventFactories: InitialEventFactoriesTuple,
+  ): SwarmProtocol<SwarmProtocolName, MachineEvent.Factory.Reduce<InitialEventFactoriesTuple>> => {
+    const eventFactories = MachineEvent.Factory.convertTupleToArray(registeredEventFactories)
```

Comment:
This is the only place where `convertTupleToArray` is used and it is only in the implementation of this function, not in the signature. I tried removing it and using `any` inside this function and everything compiled. Is this assessment correct? In that case, the reason for this utility function is only to make this `make` implementation type-safe, right?

Reply:
You are quite right. It might have been the change to `Reduce`.

Reply:
fixed in https://github.com/Actyx/machines/pull/41/commits/a1f2e1537b83ed1ba115bb33aba3bc1136eaa849